File: app/src/main/java/com/jupiter/tusaandroid/MainActivity.java
package com.jupiter.tusaandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.jupiter.tusaandroid.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Used to load the 'tusaandroid' library on application startup.
    static {
        System.loadLibrary("tusaandroid");
    }

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // Example of a call to a native method
        TextView tv = binding.sampleText;
        tv.setText(stringFromJNI());
    }

    /**
     * A native method that is implemented by the 'tusaandroid' native library,
     * which is packaged with this application.
     */
    public native String stringFromJNI();
}

File: app/src/main/java/com/example/tusa_android/MainActivity.kt
package com.example.tusa_android

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.tusa_android.cache.MemoryBitmapsCache
import com.example.tusa_android.my_profile.MyProfile
import com.example.tusa_android.network.AppUrl
import com.example.tusa_android.network.Authentication
import com.example.tusa_android.network.Grpc
import com.example.tusa_android.screen.ScreenMetricsUtils


class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Grpc.createInstance(AppUrl.grpcHost, AppUrl.grpcPort)

        val sharedPreferences = getSharedPreferences("AUTHENTICATION", Context.MODE_PRIVATE)
        Authentication.createInstance(sharedPreferences)

        MemoryBitmapsCache.createInstance(savedInstanceState)

        MyProfile.createInstance()

    }
}